clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1016,6 +1024,8 @@ public void assign(Collection<TopicPartition> partitions) {
 
             return ConsumerRecords.empty();
         } finally {
+            lastPollEndTimestamp = time.milliseconds();
```

Comment:
So we may have one extra `time.milliseconds()` for every poll. Not sure if this will be a problem. I recall that there used to be performance issue due to extra `time.millseconds()` in `consumer.poll()`. Do you remember this issue?

But it seems we have to bear with this extra call in order to have this metric.